controller/lib/core/sprintf.cpp

```diff
@@ -0,0 +1,623 @@
+/* Copyright 2020, RespiraWorks
```

Comment:
This is pretty tough to check for correctness, so I'm a little paranoid about ensuring that it doesn't get included in the final build.

- Could we add a TODO in the header to disable this module in release builds because it hasn't been reviewed/tested up to the standards of the rest of the code?
- Could we file a bug to add unit tests for this module?  Since it's not release code that doesn't need to block this PR, but it would be a good first bug.

Reply:
Good idea, will do.